diff --git a/torchvision/csrc/io/image/cpu/encode_jpeg.cpp b/torchvision/csrc/io/image/cpu/encode_jpeg.cpp
index d2ed73071a..8b13ab633c 100644
--- a/torchvision/csrc/io/image/cpu/encode_jpeg.cpp
+++ b/torchvision/csrc/io/image/cpu/encode_jpeg.cpp
@@ -19,7 +19,7 @@ torch::Tensor encode_jpeg(const torch::Tensor& data, int64_t quality) {
     (JPEG_LIB_VERSION_MAJOR == 9 && JPEG_LIB_VERSION_MINOR <= 2)
 using JpegSizeType = unsigned long;
 #else
-using JpegSizeType = size_t;
+using JpegSizeType = unsigned long;
 #endif
 
 using namespace detail;
