diff --git a/torchvision/csrc/io/image/cpu/decode_png.cpp b/torchvision/csrc/io/image/cpu/decode_png.cpp
index ede14c1e94..65079a26f1 100644
--- a/torchvision/csrc/io/image/cpu/decode_png.cpp
+++ b/torchvision/csrc/io/image/cpu/decode_png.cpp
@@ -2,6 +2,7 @@
 #include "../common.h"
 #include "common_png.h"
 #include "exif.h"
+#include <torch/types.h>
 
 namespace vision {
 namespace image {
@@ -197,7 +198,7 @@ torch::Tensor decode_png(
   auto is_16_bits = bit_depth == 16;
   auto tensor = torch::empty(
       {int64_t(height), int64_t(width), channels},
-      is_16_bits ? at::kUInt16 : torch::kU8);
+      is_16_bits ? c10::kUInt16 : torch::kU8);
   if (is_little_endian()) {
     png_set_swap(png_ptr);
   }
