diff --git a/torchvision/csrc/io/image/cpu/decode_png.cpp b/torchvision/csrc/io/image/cpu/decode_png.cpp
index ede14c1e94..a42bd06c35 100644
--- a/torchvision/csrc/io/image/cpu/decode_png.cpp
+++ b/torchvision/csrc/io/image/cpu/decode_png.cpp
@@ -195,9 +195,13 @@ torch::Tensor decode_png(
 
   auto num_pixels_per_row = width * channels;
   auto is_16_bits = bit_depth == 16;
+
+  if (is_16_bits) {
+    throw std::runtime_error("16-bit PNG images are not supported.");
+  }
+
   auto tensor = torch::empty(
-      {int64_t(height), int64_t(width), channels},
-      is_16_bits ? at::kUInt16 : torch::kU8);
+      {int64_t(height), int64_t(width), channels}, torch::kU8);
   if (is_little_endian()) {
     png_set_swap(png_ptr);
   }
diff --git a/torchvision/csrc/ops/nms.cpp b/torchvision/csrc/ops/nms.cpp
index 5ecf8812f1..2cbbe93873 100644
--- a/torchvision/csrc/ops/nms.cpp
+++ b/torchvision/csrc/ops/nms.cpp
@@ -18,11 +18,5 @@ at::Tensor nms(
   return op.call(dets, scores, iou_threshold);
 }
 
-TORCH_LIBRARY_FRAGMENT(torchvision, m) {
-  m.set_python_module("torchvision._meta_registrations");
-  m.def(TORCH_SELECTIVE_SCHEMA(
-      "torchvision::nms(Tensor dets, Tensor scores, float iou_threshold) -> Tensor"));
-}
-
 } // namespace ops
 } // namespace vision
